core/packages/contracts/src/InboundQueue.sol

```diff
@@ -51,15 +49,14 @@ contract InboundQueue is AccessControl {
     event RewardUpdated(uint256 reward);
     event GasToForwardUpdated(uint256 gasToForward);
 
-
     error InvalidProof();
     error InvalidNonce();
-    error InvalidHandler();
+    error InvalidRecipient();
     error NotEnoughGas();
 
-    constructor(IParachainClient _parachainClient, IVault _vault, uint256 _reward) {
-        _grantRole(ADMIN_ROLE, msg.sender);
-        _setRoleAdmin(ADMIN_ROLE, ADMIN_ROLE);
+    constructor(Registry registry, IParachainClient _parachainClient, Vault _vault, uint256 _reward)
+        RegistryLookup(registry)
```

Comment:
Also need to add construct `Registry()` explicit construct from parent.

Reply:
I think you mean an explicit call to `RegistryLookup` constructor, which I am doing.